### ISSUE TITLE
conflunent key

### ISSUE THREAD

**Issue by 13639577519** [rtdi/S4HanaConnector#21]
Hello, I'm sorry to disturb you. I'm using conform to dock with s4hanaconnector, but as a beginner of conform, I can't find where you need to find these parameters of key in your page. Can you give me a suggestion on where to view these parameters? I think you and I have never set a key in the process of deploying confidant

![image](https://user-images.githubusercontent.com/53895233/111027721-f643d480-842c-11eb-9ea6-9242e58fa53f.png)

**Comment by wernerdaehn** [comment]
In this screen you enter the connection to a Kafka server. The only mandatory parameter for connecting to Kafka is the bootstrap server list.
https://jaceklaskowski.gitbooks.io/apache-kafka/content/kafka-properties-bootstrap-servers.html
This value is provided to you by whoever operates the Kafka cluster, in my case I am using Confluent cloud and get the bootstrap server list from there.

Very likely the server will not be open to public, so you must specify a user and a password. In case of Kafka the username is the cluster key and the password the secret string. Plus the login method. Again all these values are provided to you by the Kafka cluster admins.

As the payload used is Avro, Kafka needs to store the schema definition somewhere. That is the schema registry. This is a Restful endpoint - schema registry url - and it might be username/password protected. Hence registry key/secret must be provided in such case. If no schema registry is entered, the connector creates a topic called _schema and is using that for storage.

https://docs.confluent.io/cloud/current/client-apps/api-keys.html

**Comment by 13639577519** [comment]
Thank you for your reply. According to what you said, only the bootstrap server list is required, and the rest of the parameters are optional, right? I can't connect to Kafka even though I only filled in the bootstrap server. The abnormal screenshot is as follows. Can you see the reason? My client is deployed on a Linux server. It is deployed on a single node without a cluster

Have you ever tried to fill in the address of the server only?
![image](https://user-images.githubusercontent.com/53895233/111065286-df1bea00-84f3-11eb-9716-1b3c3407b50a.png)
![image](https://user-images.githubusercontent.com/53895233/111065375-476acb80-84f4-11eb-812c-ded8487af602.png)

**Comment by 13639577519** [comment]
After I fill in the schema registry URL, I display restful call got an error


![image](https://user-images.githubusercontent.com/53895233/111068116-5eb0b580-8502-11eb-9178-dea378e8f472.png)
![image](https://user-images.githubusercontent.com/53895233/111068148-8738af80-8502-11eb-8997-235823336dca.png)

**Comment by wernerdaehn** [comment]
Both points to me to a permission issue. But let me install a Single-Node cluster today and I give it a try myself.

**Comment by wernerdaehn** [comment]
It really depends on your environment. Where is Kafka running, where is the s4hanaconnector running etc.

To cover most cases I did the following steps:
For reference, the Kafka docker install is listed under (Confluent Community Edition Wiki)[https://docs.confluent.io/platform/current/quickstart/cos-docker-quickstart.html], the Step 1.
1. Downloaded the docker compose script: `curl --silent --output docker-compose.yml \
  https://raw.githubusercontent.com/confluentinc/cp-all-in-one/6.1.0-post/cp-all-in-one-community/docker-compose.yml`
2. Modified the configuration so that all components internally talk via localhost:29092 but the external client is using the correct hostname. That meant replacing the line in the yml file. From `KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://broker:29092,PLAINTEXT_HOST://localhost:9092` to`KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://broker:29092,PLAINTEXT_HOST://hana2:9092` where "hana2" is the hostname of the Linux server. Yes, my Linux server runs Hana Express and all docker images.
3. Executed `docker-compose up -d` to startup all, zookeeper, broker and schema registry (and more).
4. Validated that all is active via `docker ps -a` and `curl -X GET http://hana2:8081/subjects/`. The latter queries the schema registry and it stores all in Kafka, so it is an end-to-end test.

Case 1 - S4HanaConnector is somewhere else

6. Opened the s4hanaconnector webpage
5. Entered `hana2:9092` as bootstrap server and `http://hana2:8081/` as schema registry URL. All keys and secrets are empty.
6. Clicked on save and it succeeded.
7. Went back to the Homepage of the s4hanaconnector and clicked on server topics to see the five default topics created (Kafka internal topics like _schemas are filtered)

Case 2 - S4HanaConnector is on the same box

Here the problem is that docker is shielding all containers from each other. So if you would go into the running s4hanaconnector and execute above `curl -X GET http://hana2:8081/subjects/` it would fail as hana2 as hostname is not known. localhost is no solution either as it mean all network services within the contain, not the host's localhost.

6. Join the running s4hanaconnector to the docker-network created for Kafka: `docker network connect downloads_default s4hanaconnector`
7. Enter `broker:29092` as bootstrap servers. "broker" is the host of the Kafka container within the downloads_default network and we configured in the docker.yml that internal traffic is routed via the port 29092.
8. Enter `http://schema-registry:8081` as schema registry url. Same story, the "schema-registry" is the hostname of the schema registry within the downloads_default network.
9. Clicked on save and it succeeded.
10. Went back to the Homepage of the s4hanaconnector and clicked on server topics to see the five default topics created (Kafka internal topics like _schemas are filtered)

As you can see it is all a docker config issue. I did not have those problems because I am using a remote Kafka installation, one Confluent Cloud provided. Hence I just copied the values they provided me with to configure my local s4hanaconnector.

**Comment by 13639577519** [comment]
Thank you very much for your reply. At the same time, thank you very much for testing in your busy schedule. I will check the permissions according to the information you provided, or try to deploy according to your method. Thank you again

**Comment by 13639577519** [comment]
I have another question to ask you. Is your Kafka deployed in docker or on the server running docker?

**Comment by wernerdaehn** [comment]
Not sure I get your point. In my case I have a Linux server with docker and there I installed/run (docker pull, docker run) the confluent docker image as one container and the s4hanaconnector as another container.
In addition my development windows computer runs the s4hanaconnector as well but without docker.

**Comment by 13639577519** [comment]
Hello, I have to disturb you again. According to your steps, I tried. Now the error report is that I can't create topic, but I couldn't read it before. Do you know how to deal with this problem?

![Uploading image.png…]()

**Comment by 13639577519** [comment]
![image](https://user-images.githubusercontent.com/53895233/111299933-7cf5ed00-868b-11eb-86b7-836a968a38ed.png)

**Comment by 13639577519** [comment]
Or you know that this problem is caused by a specific parameter in the configuration. Please let me know, please, I have been stuck in the pipeline connection interface of Kafka for a long time

**Comment by wernerdaehn** [comment]
I would like to see the webapp log by going into the docker image. Do you know how to do that?
Plan B: The home page of the connector has a "support info" tile and within it shows the application log.
Plan C is a web meeting.

**Comment by 13639577519** [comment]
I don't know if you are talking about this log
![image](https://user-images.githubusercontent.com/53895233/111314524-73c14c00-869c-11eb-9687-a63e732e6fbb.png)

**Comment by 13639577519** [comment]
I don't know if you are talking about this log

![image](https://user-images.githubusercontent.com/53895233/111314766-ac612580-869c-11eb-9a6f-97359c5302cc.png)

**Comment by wernerdaehn** [comment]
Yes, but that's just a shut down http connection, so does not help.

On the computer running the s4hanaconnector, execute the command `docker exec -it s4hanaconnector bash`. 
Then `cd logs` to get into the webapp log folder and `cat webapp.log`. That is the information I am looking for.

And my guess: It says something about adminclient and localhost. But we will see.

**Comment by 13639577519** [comment]
Where is this parameter defined in the configuration file?
![image](https://user-images.githubusercontent.com/53895233/111326305-69587f80-86a7-11eb-90fe-4878bcb574b5.png)

**Comment by wernerdaehn** [comment]
You never clicked on "save" when typing in the properties in the Kafka screen?

**Comment by 13639577519** [comment]
Every time I click "save", only when I click "save" can I see the error

![image](https://user-images.githubusercontent.com/53895233/111332631-ecc89f80-86ac-11eb-8196-e4296856c11f.png)

**Comment by wernerdaehn** [comment]
Web meeting?

**Comment by 13639577519** [comment]
I'm really sorry, my office network does not support the use of video software, do you have any contact information? I'll add you

**Comment by wernerdaehn** [comment]
werner.daehn@rtdi.io

**Comment by 13639577519** [comment]
OK, I'll re deploy my environment, and then sort out the situation and problems and send them to you by email
